icu4c/source/tools/toolutil/pkg_genc.cpp

```diff
@@ -435,6 +436,18 @@ writeCCode(
         uprv_strcpy(outFilePath, buffer);
     }
 
+    if(optEntryPoint != NULL) {
+        uprv_strcpy(entry, optEntryPoint);
+        uprv_strcat(entry, "_dat");
+    }
```

Comment:
This is good, but logically, it should move down a few lines, between opening `out` and the existing "dashes or dots" loop.